tests/functional/event_handler/_pydantic/test_openapi_validation_middleware.py

```diff
@@ -1983,3 +2259,131 @@ def get_user(user_id: int) -> UserModel:
+
+    def _validate_powertools(value: str) -> str:
+        if not value.startswith("Powertools"):
+            raise ValueError("Full name must start with 'Powertools'")
+        return value
+
+    class QuerySimple(BaseModel):
+        full_name: Annotated[str, StringConstraints(min_length=5), AfterValidator(_validate_powertools)]
+        next_token: Base64UrlStr
+        search_id: str
+
+    @app.get("/query-model-simple")
+    def query_model(params: Annotated[QuerySimple, Query()]) -> dict[str, Any]:
+        return {
+            "fullName": params.full_name,
+            "nextToken": params.next_token,
+            "searchId": params.search_id,
+        }
+
+    class QueryAdvanced(BaseModel):
+        full_name: Annotated[str, StringConstraints(min_length=5)]
+        next_token: str
+        search_id: Annotated[str, Field(alias="id")]  # Using str instead of UUID4 for simpler testing
+
+        model_config = ConfigDict(
+            alias_generator=alias_generators.to_camel,
+            validate_by_alias=True,
+            serialize_by_alias=True,
+        )
+
+    @app.get("/query-model-advanced")
+    def query_model_advanced(params: Annotated[QueryAdvanced, Query()]) -> dict[str, Any]:
+        return params.model_dump()
+
+    # Test QuerySimple with validators
+    gw_event["path"] = "/query-model-simple"
+    gw_event["queryStringParameters"] = {
+        "full_name": "Powertools Lambda",
+        "next_token": "dGVzdA==",  # base64url encoded "test"
+        "search_id": "search-123",
+    }
+
+    result = app(gw_event, {})
+    assert result["statusCode"] == 200
+
+    body = json.loads(result["body"])
+    assert body["fullName"] == "Powertools Lambda"
+    assert body["nextToken"] == "test"
+    assert body["searchId"] == "search-123"
+
+    # Test QuerySimple validation error (name doesn't start with "Powertools")
+    gw_event["queryStringParameters"] = {
+        "full_name": "Lambda Powertools",  # Wrong order
+        "next_token": "dGVzdA==",
+        "search_id": "search-123",
+    }
+
+    result = app(gw_event, {})
+    assert result["statusCode"] == 422
+
+    body = json.loads(result["body"])
+    assert "detail" in body
+    errors = body["detail"]
+
+    # Test QueryAdvanced with ConfigDict and alias_generator
+    gw_event["path"] = "/query-model-advanced"
+    gw_event["queryStringParameters"] = {
+        "fullName": "Advanced Test",  # camelCase from alias_generator
+        "nextToken": "dGVzdA==",  # camelCase from alias_generator
+        "id": "search-456",  # explicit alias
+    }
+
+    result = app(gw_event, {})
+    assert result["statusCode"] == 200
+
+    body = json.loads(result["body"])
+    # Should return with camelCase keys due to serialize_by_alias=True
+    assert body["fullName"] == "Advanced Test"
+    assert body["nextToken"] == "dGVzdA=="
+    assert body["id"] == "search-456"
+
+    # Test QueryAdvanced with snake_case field names (should also work due to populate_by_name behavior)
```

Comment:
if you want to test `populate_by_name` behavior you either need to use `populate_by_name` or set `validate_by_name` **and** `validate_by_alias`.